src/mongoose_hooks.erl

```diff
@@ -1,76 +1,81 @@
 -module(mongoose_hooks).
 
--export([auth_failed/2,
-         c2s_broadcast_recipients/6,
-         c2s_filter_packet/6,
+-export([auth_failed/3,
+         c2s_broadcast_recipients/7,
+         c2s_filter_packet/7,
          c2s_preprocessing_hook/3,
          c2s_presence_in/5,
-         c2s_stream_features/1,
-         c2s_unauthenticated_iq/4,
+         c2s_stream_features/2,
+         c2s_stream_features/3,
+         c2s_unauthenticated_iq/5,
          c2s_update_presence/2,
-         check_bl_c2s/2,
+         check_bl_c2s/3,
          failed_to_store_message/4,
          forbidden_session_hook/3,
          offline_groupchat_message_hook/5,
          offline_message_hook/5,
          presence_probe_hook/5,
-         privacy_check_packet/6,
-         privacy_get_user_list/3,
+         privacy_check_packet/7,
+         privacy_get_user_list/4,
          privacy_iq_get/6,
          privacy_iq_set/5,
          privacy_updated_list/4,
-         resend_offline_messages_hook/3,
-         roster_get_subscription_lists/3,
-         roster_get_versioning_feature/1,
+         resend_offline_messages_hook/4,
+         roster_get_subscription_lists/4,
+         roster_get_versioning_feature/2,
+         roster_get_versioning_feature/3,
          roster_in_subscription/7,
-         roster_out_subscription/5,
+         roster_out_subscription/6,
          session_opening_allowed_for_user/3,
-         set_presence_hook/5,
+         set_presence_hook/6,
          sm_broadcast/6,
          sm_filter_offline_message/5,
          sm_register_connection_hook/4,
          sm_remove_connection_hook/6,
          unacknowledged_message/3,
-         unset_presence_hook/5,
+         unset_presence_hook/6,
          user_available_hook/3,
          user_receive_packet/6,
          user_sent_keep_alive/2,
          user_send_packet/5,
          xmpp_bounce_message/2,
          xmpp_send_element/3]).
 
--spec auth_failed(Server, Username) -> Result when
-    Server :: jid:server(),
+-spec auth_failed(HookServer, Username, Server) -> Result when
```

Comment:
is it really possible to have `Server` non-equal to `HookServer`?

Reply:
In theory it's possible. This change aims to provide more clarity on what arguments are passed to the handlers. Currently the idea is that the first argument which is always `HookServer` is used to find the hook handlers. Then the second argument is most often the accumulator and the rest is passed to the handler.

Initially there was only one `Server` but we thought it makes writing handlers less clear.